File: Source/Aladdin.CAPI.ANSI/ASN1/ANSI/X962/ECCSupplements.cs
namespace Aladdin.ASN1.ANSI.X962
{
    ////////////////////////////////////////////////////////////////////////////////
    // ECCSupplements ::= CHOICE {
    //      namedMultiples      [0] EXPLICIT NamedMultiples,
    //      specifiedMultiples  [1] EXPLICIT SpecifiedMultiples
    // }
    ////////////////////////////////////////////////////////////////////////////////
    public class ECCSupplements : Choice
    {
	    // информация о структуре
	    private static readonly ObjectInfo[] info = new ObjectInfo[] { 

		    new ObjectInfo(new ObjectCreator<NamedMultiples    >().Factory(), Cast.E, Tag.Context(0)), 
		    new ObjectInfo(new ObjectCreator<SpecifiedMultiples>().Factory(), Cast.E, Tag.Context(1)) 
	    }; 
	    // конструктор
	    public ECCSupplements() : base(info) {} 
    }
}

File: Source/Aladdin.CAPI.ANSI.Java/src/aladdin/asn1/ansi/x962/ECCSupplements.java
package aladdin.asn1.ansi.x962;
import aladdin.asn1.*; 

////////////////////////////////////////////////////////////////////////////////
// ECCSupplements ::= CHOICE {
//      namedMultiples      [0] EXPLICIT NamedMultiples,
//      specifiedMultiples  [1] EXPLICIT SpecifiedMultiples
// }
////////////////////////////////////////////////////////////////////////////////
public final class ECCSupplements extends Choice
{
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(NamedMultiples    .class).factory(), Cast.E, Tag.context(0)), 
		new ObjectInfo(new ObjectCreator(SpecifiedMultiples.class).factory(), Cast.E, Tag.context(1)) 
	}; 
	// конструктор
	public ECCSupplements() { super(info); } 
}
